config/routes.rb

```diff
@@ -3,28 +3,27 @@
 Rails.application.routes.draw do
   def draw_routes(prefix)
     scope "#{prefix}/#{Settings.app_name}" do
-      namespace :v1 do
-        resources :benchmarks, only: [:index, :show]
-        resources :business_objectives, only: [:index, :show]
-        resources :profiles do
-          member do
-            get 'tailoring_file'
+      concern :rest_api_v1 do
+        scope module: 'v1' do
```

Comment:
I hope this doesn't come back to bite us. Is this some newer way to define namespaces, or?

Reply:
Newer way is to use `namespace :v1`, but it affects link generators (`_url` methods) for endpoints outside of the `/v1` path.

Reply:
So what's the difference exactly? Today we have (i.e.) `profiles_url` and `v1_profiles_url`.

Reply:
The `scope module: 'v1'` is necessary to use `V1` module controllers. The previous `namespace :v1` did set the path (`/v1`) as well as the module (`V1`). It is not possible to use it for the concern, as it could not be set on toplevel path.

`profiles_url` points to `/profiles` and `v1_profiles_url` points to `/v1/profiles`, as before. With `namespace` there would not be the `profiles_url`.

Reply:
Ok, after looking at this some more, this seems like the right way to do it :) Sorry for the noise! It's much better now with the concern.